src/videojs-hls.js

```diff
@@ -540,6 +542,11 @@ videojs.HlsHandler.prototype.setCurrentTime = function(currentTime) {
     return currentTime;
   }
 
+  // if we are in the middle of appending a segment, let it finish up
+  if (this.pendingSegment_ && this.pendingSegment_.buffered) {
+    return currentTime;
```

Comment:
The issue is that the following code cancels the segment XHR and  clears the `this.pendingSegment_` property. When 'updateend' is called we have no way of annotating segments. This was resulting in us getting the same segment (the first) twice during a live playlist start.

The steps went something like:
* We would start the request for the right segment via checkBuffer_ (based on currentTime)
* The seek handler would try to cancel it but it was already in flight
* We would finish appending the segment but store no information about it in `updateend`
* Then we would fetch the same segment (based on currentTime)
* In that second segment's `updateend`, we would record *no* new information because the buffer already has the same segment!

It was a strange and difficult to track down situation.